lib/content-services/src/lib/content-node-selector/content-node-selector.component.ts

```diff
@@ -47,21 +47,24 @@ export class ContentNodeSelectorComponent {
 
     onSelect(nodeList: Node[]) {
         this.chosenNode = nodeList;
-        this.updateTitle(nodeList);
+    }
+
+    onSiteChange(siteTitle: string) {
+        this.updateTitle(siteTitle);
     }
 
     onClick(): void {
         this.data.select.next(this.chosenNode);
         this.data.select.complete();
     }
 
-    updateTitle(nodeList: Node[]): void {
-        if (this.action === 'CHOOSE' && nodeList) {
-            this.title = this.getTitleTranslation(this.action, nodeList[0].name);
+    updateTitle(siteTitle: string) {
+        if (this.action === 'CHOOSE' && siteTitle) {
+            this.title = this.getTitleTranslation(this.action, siteTitle);
         }
```

Comment:
@adinapitul for your concerns about breaking the title in the Copy / Move dialog. We are checking here that the action is "CHOOSE" (action for upload dialog) and not "COPY" or "MOVE" before updating the title.